src/Sylius/Bundle/CoreBundle/spec/EventListener/Workflow/Order/CreatePaymentListenerSpec.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\CoreBundle\EventListener\Workflow\Order;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Abstraction\StateMachine\StateMachineInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Payment\PaymentTransitions;
+use Symfony\Component\Workflow\Event\CompletedEvent;
+use Symfony\Component\Workflow\Marking;
+
+final class CreatePaymentListenerSpec extends ObjectBehavior
+{
+    function let(StateMachineInterface $compositeStateMachine): void
+    {
+        $this->beConstructedWith($compositeStateMachine);
+    }
+
+    function it_throws_an_exception_on_non_supported_subject(\stdClass $callback): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('__invoke', [new CompletedEvent($callback->getWrappedObject(), new Marking())]);
+    }
+
+    function it_does_nothing_if_payment_cannot_be_created(
+        StateMachineInterface $compositeStateMachine,
+        OrderInterface $order,
+        PaymentInterface $payment,
+    ): void {
+        $event = new CompletedEvent($order->getWrappedObject(), new Marking());
+        $order->getPayments()->willReturn(new ArrayCollection([$payment->getWrappedObject()]));
+        $compositeStateMachine
+            ->can($payment, PaymentTransitions::GRAPH, PaymentTransitions::TRANSITION_CREATE)
+            ->willReturn(false)
+        ;
+
+        $this($event);
+
+        $compositeStateMachine
+            ->apply($payment, PaymentTransitions::GRAPH, PaymentTransitions::TRANSITION_CREATE)
+            ->shouldNotHaveBeenCalled()
+        ;
+    }
+
+    function it_creates_payments(
```

Comment:
```suggestion
    function it_marks_payments_as_created(
```
or
```suggestion
    function it_marks_payments_as_new(
```
or
```suggestion
    function it_applies_transistion_create_on_payments(
```
Naming's a bit messy in this whole scope, I'd expect `it_creates_payments` to be a factory test.